src/main/java/edu/agh/wfiis/designpatterns/patternsUsageExample/design_patterns/BuilderDesignPattern.java

```diff
@@ -0,0 +1,30 @@
+package edu.agh.wfiis.designpatterns.patternsUsageExample.design_patterns;
+
+public class BuilderDesignPattern extends DesignPattern {
+    BuilderDesignPattern(){}
+
+    @Override
+    public String getName() {
+        return "Builder";
+    }
+
+    @Override
+    public Type getType() {
+        return Type.CREATIONAL;
+    }
+
+    @Override
+    public String getDescription() {
+        return "some builder desc";
```

Comment:
In my view the design pattern metadata like description and example should not be hard-coded in class definition, this doesn't solve the telescoping constructor problem, it moves the problem somewhere else in the code.